src/Catel.MVVM/Catel.MVVM.Shared/Collections/SuspensionContext.cs

```diff
@@ -165,32 +154,22 @@ public bool TryRemoveItemFromOldItems(int index, T item)
         /// <summary>
         /// Tries to remove the item from new items
         /// </summary>
-        /// <param name="index">
-        /// The item index.
-        /// </param>
         /// <param name="item">
         /// The item.
         /// </param>
         /// <returns>
         /// <c>true</c> if removed, otherwise <c>false</c>.
         /// </returns>
-        public bool? TryRemoveItemFromNewItems(int index, T item)
+        public bool? TryRemoveItemFromNewItems(T item)
         {
             if (Mode == SuspensionMode.None || Mode == SuspensionMode.Mixed)
             {
                 var newIdx = NewItems.LastIndexOf(item);
```

Comment:
List.LastIndexOf has O(n). As this method is called upon .Remove which has O(n) this results in 2xO(n) for removing one item.